### PR TITLE
feat(i18n): store article translation to db instead of cache in redis

### PR DESCRIPTION
**Pull request by robertu7**

### CODE REVIEW

**Review by guoliu — APPROVED** [review]

File: src/queries/article/translation.ts [hunk 1]
```diff
@@ -23,12 +39,30 @@ const resolver: ArticleToTranslationResolver = async (
...
+      title,
+      content,
+      summary,
+      language: target,
+    }
+    await atomService.upsert({
+      table: 'article_translation',
+      where: { articleId },
+      create: data,
+      update: { ...data, updatedAt: new Date() },
```
Comment (tx0c) [inline src/queries/article/translation.ts:55]: updatedAt: this.knex.fn.now(),

File: src/queries/article/translation.ts [hunk 2]
```diff
@@ -23,12 +39,30 @@ const resolver: ArticleToTranslationResolver = async (
     )
```
Comment (tx0c) [inline src/queries/article/translation.ts:39]: does GCP charge by count or by text length? and do we translate the tags, and fix the translation length limit issue; `四萬字長文` seems always fail

although can be in a separate EPIC

Comment (robertu7) [reply src/queries/article/translation.ts:39]: > does GCP charge by count or by text length? and do we translate the tags

[count by characters](https://cloud.google.com/translate/pricing), seems good to translate tags as well.


> and fix the translation length limit issue; 四萬字長文 seems always fail

“The Cloud Translation API is optimized for translating smaller requests. The recommended maximum length for each request is 5K characters ([code points](https://wikipedia.org/wiki/UTF-8)). However, the more characters that you include, the higher the response latency. For Cloud Translation - Advanced, the maximum number of code points for a single request is 30K. Cloud Translation - Basic has a maximum request size of 100K bytes. An alternative option is to use batch translations to send more content per request as a long running operation.”
https://cloud.google.com/translate/quotas#content

We can use `batchTranslateText`, will fix in this PR.

Comment (robertu7) [reply src/queries/article/translation.ts:40]: `batchTranslateText` is more complex to call, shall take another time to fix [it](https://github.com/thematters/matters-server/issues/2694);

**Review by tx0c — APPROVED** [review]
looks big enough to merge; don't try to solve too many things in one PR

**Pull request merged**